lib/ads/client_side_ad_manager.js

```diff
@@ -452,7 +457,9 @@ shaka.ads.ClientSideAdManager = class {
         (new Map()).set('originalEvent', e)));
     if (this.ad_ && this.ad_.isLinear()) {
       this.adContainer_.removeAttribute('ad-active');
-      this.video_.play();
+      if (!this.contentCompletedCalled_) {
```

Comment:
Couldn't you also check `this.video_.ended`? That wouldn't require adding a new local variable.

Reply:
Works great with `this.video_.ended` and way cleaner. PR updated.